### ISSUE TITLE
Some convenient syntax for field reflection

### ISSUE THREAD

**Issue by nikomatsakis** [nikomatsakis/fields-in-traits-rfc#2]
The RFC does not offer any convention way to extract a "getter" function. It was [proposed by @petrochenkov](https://github.com/rust-lang/rfcs/pull/1546#r56406889) that `Trait::field_name` could be "sugar" for a getter.

There are some questions that arise, though. For example, would this apply only to `Copy` types? Any `Clone` type? `Clone` is not currently a "lang-item" *exactly*, though it is the supertrait of `Copy`, so it is certainly "part of the language". I guess one could imagine that, for every field `field` of type `FieldType`, we have a "getter" roughly equivalent to:

```rust
fn field(&self) -> FieldType where FieldType: Clone {
    self.field.clone()
}
```

In other words, if you use `Trait::field` for some field where `FieldType` is not `Clone`, you would get an error at that point (since the where clause is not satisfied.

**Comment by burdges** [comment]
Why wouldn't `foo.map(_.method(_))` be `|x| foo.map(|y| x.method(y))` or whatever?  It just stops at the first `Fn` type?  I think haskell's operator and `$` syntax avoids this confusion.
